### PR TITLE
Unlock content immediately for posts with the subscriber access level

### PR DESCRIPTION
**Pull request by romarioraffington**
Closes  https://github.com/Automattic/wp-calypso/issues/69664
Closes https://github.com/Automattic/wp-calypso/issues/70119

#### Changes proposed in this Pull Request:

We do not immediately unlock a post with the subscriber access level. Instead, we send a verification email. This is not an ideal user experience. This PR gives immediate access to subscribers after subscribing once the post has the subscriber access level.

This PR and patch will also hide the free option in the `screen-choose-plan.php` template if the post is marked for paid subscribers.

#### Testing instructions:
See D92672-code

### CODE REVIEW

**Review by artpi — CHANGES_REQUESTED** [review]
I see where this approach is going.

You tapped into the return of the current subscribe block flow to capture these parameters and unlock the content.
This will work short-term, but won't work with setting up the cookie for when somebody navigates from the site and back.

The flow would work like so:

1. Click on subscribe
2. Modal opens
3. Choose free option
4. Modal closes
5. The old subscribe form submits
6. Redirects to WPCOM
7. Subscribes user, sends confirmation email
8. Redirects back with query string
9. This PR is intercepting this query string and unlocking content based on it's contents.

What I think should be happening:

1. Click on subscribe
2. Modal opens
3. Choose free option
4. This PR should change how this free option behaves. It should subscribe user inside the modal and redirect with a valid JWT token (with blog_sub=confirming) that in turn will unlock content

**Comment by artpi** [comment]
I'm open to discussion!

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php [hunk 1]
```diff
@@ -71,8 +74,21 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
 		}
 
 		if ( $is_valid_token ) {
-			if ( 'subscribers' === $access_level ) {
-				return 'active' === $payload['blog_sub'];
+			if ( self::POST_ACCESS_LEVEL_SUBSCRIBERS === $access_level ) {
```
Comment (millerf) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php]: I have modified all that logic in [this PR](https://github.com/Automattic/jetpack/pull/27352). Do you mind to wait and rebase on it? It should be minimal work compared to the other way around ....

**Review by millerf — APPROVED** [review]
LGTM- added a comment about merging

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php [hunk 2]
```diff
@@ -71,8 +74,21 @@ public function visitor_can_view_content( $valid_plan_ids, $access_level ) {
...
+				/**
+				 * Allow access to the content if:
+				 *
+				 * Active: user has a valid subscription
+				 * Pending: user has subscribed for a free plan
+				 */
+				return in_array(
```
Comment (artpi) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php]: Yes, that is what I was thinking about!

**Pull request merged**